src/main/java/org/jabref/Globals.java

```diff
@@ -90,10 +92,13 @@ public static void startBackgroundTasks() {
     }
 
     private static void stopTelemetryClient() {
-        if (Globals.prefs.shouldCollectTelemetry()) {
-            getTelemetryClient().ifPresent(client -> client.trackSessionState(SessionState.End));
-            getTelemetryClient().ifPresent(client -> client.flush());
-        }
+        getTelemetryClient().ifPresent(client -> {
+            client.trackSessionState(SessionState.End);
+            client.flush();
+
+            // Workaround for bug https://github.com/Microsoft/ApplicationInsights-Java/issues/662
```

Comment:
Add FIXME so we can revert this after they merged the bug fix.

Reply:
I really would use my PR as a base. As this done does more than the initial necessary solution.

Reply:
I don't understand the complication (including merge conflicts) but well it doesn't really matter. @Siedlerchr @stefan-kolb can I merge this PR?

Reply:
Guess so.